src/dmd/globals.d

```diff
@@ -148,7 +148,7 @@ extern (C++) struct Param
     bool betterC;           // be a "better C" compiler; no dependency on D runtime
     bool addMain;           // add a default main() function
     bool allInst;           // generate code for all template instantiations
-    bool fix16997;          // fix integral promotions for unary + - ~ operators
+    FeatureState fix16997;          // fix integral promotions for unary + - ~ operators
```

Comment:
Where is the default setting done?

Reply:
https://github.com/dlang/dmd/blob/03d55018cff4bb3c4d75c7e4e356e393b8b37821/src/dmd/globals.d#L94

Reply:
That's not a good default if the intent is always enabled.

Reply:
I did not create the FeatureState enum in this PR, I just used an existing system that works for the DIP25 switch and use that for the easy creation of the revert switch. I believe that @atilaneves is the one who created the FeatureState enum, in the first place. Regardless I am not planing to change the FeatureState enum  here as it is out of scope of this PR.

Reply:
But _why_ are you using FeatureState? Nothing is forcing you to change the field type of `fix16997` here.

All you need to do is: `bool fix16997 = true;`.

Reply:
> But why are you using FeatureState?

@adamdruppe wants a revert switch here. FeatureStates allows me to set it to default while creating a revert switch easily.
No need to deprecate the preview switch here.

Reply:
> > But why are you using FeatureState?
> 
> @adamdruppe wants a revert switch here. FeatureStates allows me to set it to default while creating a revert switch easily.
> No need to deprecate the preview switch here.
> 

But this isn't a tri-state option, and FeatureSet isn't needed for revert to work, so I'm still stumped as to why this is necessary.